models/issue.go

```diff
@@ -1569,6 +1569,12 @@ func SearchIssueIDsByKeyword(kw string, repoIDs []int64, limit, start int) (int6
 }
 
 func updateIssue(e Engine, issue *Issue) error {
+	if issue.PosterID <= 0 {
+		_, err := e.Table("`issue`").Where("id = ?", issue.ID).Select("poster_id").Get(&issue.PosterID)
```

Comment:
Please stick a log.Error after this if the posterID becomes > -1 with a comment to say that the calling methods need to be fixed and suggest stack trace. You should include the issue.ID.

If the posterID is still negative - then I think we should consider a log.Critical as a posterID<0 means a data loss has occurred at some point. That error message will definitely require the issue.ID in the message. (I'm not sure about Critical here but it's a definitely an error and the server owner needs to know about it.)